pulp_python/app/tasks/repair.py

```diff
@@ -0,0 +1,86 @@
+import logging
+import uuid
+from gettext import gettext as _
+
+from django.db.models.query import QuerySet
+from pulpcore.plugin.constants import TASK_STATES
+from pulpcore.plugin.models import ProgressReport
+
+from pulp_python.app.models import PythonPackageContent, PythonRepository
+from pulp_python.app.utils import artifact_to_python_content_data
+
+log = logging.getLogger(__name__)
+
+
+def repair(repository_pk: uuid.UUID) -> None:
+    """
+    Repairs metadata of all packages for the specified repository.
+
+    Args:
+        repository_pk (uuid.UUID): The primary key of the repository to repair.
+
+    Returns:
+        None
+    """
+    repository = PythonRepository.objects.get(pk=repository_pk)
+
+    log.info(
+        _(
+            "Repairing packages' metadata for the latest version of repository {}."
+        ).format(repository.name)
+    )
+    content_set = repository.latest_version().content.values_list("pk", flat=True)
+    content = PythonPackageContent.objects.filter(pk__in=content_set)
+    num_repaired = repair_metadata(content)
+    progress_report = ProgressReport(
+        message=_("Repaired packages' metadata"),
+        code="repair.metadata",
+        state=TASK_STATES.COMPLETED,
+        total=len(content_set),
+        done=num_repaired,
+    )
```

Comment:
I think this progress report should be updated as the task repairs content units. I would create this report before the repair and pass it onto the function so that the function can update the report as it goes.

Reply:
I chose to update the progress report every 100 processed packages instead of one by one, since `bulk_update` happens every 1000 updated packages. But I am not sure if this is correct.

Reply:
Have a look at the Progress reports implementation. When used with a context manager they do some sort of deferred update.